File: 2021/09/09/ReverseStr.java
/**
 * https://leetcode.com/problems/reverse-string-ii/
 * @author Andrew Jarombek
 * @since 9/9/2021
 */

public class ReverseStr {
    public String reverseStr(String s, int k) {
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i <= s.length() / k; i++) {
            int start = i * k;
            int end = Math.min(s.length(), (i + 1) * k);

            String substring = s.substring(start, end);

            if (i % 2 == 0) {
                for (int j = substring.length() - 1; j >= 0; j--) {
                    sb.append(substring.charAt(j));
                }
            } else {
                sb.append(substring);
            }
        }

        return sb.toString();
    }
}

File: 2022/03/27/reverseStr.ts
/**
 * https://leetcode.com/problems/reverse-string-ii/
 * @author Andrew Jarombek
 * @since 3/27/2022
 */

function reverseStr(s: string, k: number): string {
    let i = 0;
    let j = 0;
    let result = [];

    while (i < s.length) {
        if (j % 2 == 0) {
            for (let index = Math.min(s.length - 1, i + k - 1); index >= i; index--) {
                result.push(s.charAt(index));
            }
        } else {
            for (let index = i; index < Math.min(s.length, i + k); index++) {
                result.push(s.charAt(index));
            }
        }

        i += k;
        j++;
    }

    return result.join("");
}